sys/net/routing/ng_rpl/ng_rpl_control_messages.c

```diff
@@ -0,0 +1,582 @@
+            ng_rpl_dodag_remove(dodag);
+            return;
+        }
+
+        trickle_start(ng_rpl_pid, &dodag->trickle, NG_RPL_MSG_TYPE_TRICKLE_INTERVAL,
+                      NG_RPL_MSG_TYPE_TRICKLE_CALLBACK, (1 << dodag->dio_min),
+                      dodag->dio_interval_doubl, dodag->dio_redun);
+        dodag->version = dio->version_number;
+        ng_rpl_delay_dao(dodag);
+        parent->rank = byteorder_ntohs(dio->rank);
+        ng_rpl_parent_update(dodag, parent);
+        return;
+    }
+    else if (dodag == NULL) {
+        DEBUG("RPL: Could not allocate a new DODAG.\n");
+        if (inst->dodags == NULL) {
+            ng_rpl_instance_remove(inst);
+        }
+        return;
+    }
+
+    if (dodag->instance->mop !=
+            ((dio->g_mop_prf >> NG_RPL_MOP_SHIFT) & NG_RPL_SHIFTED_MOP_MASK)) {
+        DEBUG("RPL: invalid MOP for this instance.\n");
+        return;
+    }
+
+    if (NG_RPL_COUNTER_GREATER_THAN(dio->version_number, dodag->version)) {
+        if (dodag->node_status == NG_RPL_ROOT_NODE) {
+            dodag->version = NG_RPL_COUNTER_INCREMENT(dio->version_number);
+            trickle_reset_timer(&dodag->trickle);
+        }
+        else {
+            dodag->version = dio->version_number;
+            ng_rpl_local_repair(dodag);
+        }
+    }
+    else if (NG_RPL_COUNTER_GREATER_THAN(dodag->version, dio->version_number)) {
+        trickle_reset_timer(&dodag->trickle);
+        return;
+    }
+
+    if (dodag->node_status == NG_RPL_ROOT_NODE) {
+        if (byteorder_ntohs(dio->rank) != NG_RPL_INFINITE_RANK) {
+            trickle_increment_counter(&dodag->trickle);
+        }
+        return;
+    }
+
+    dodag->grounded = dio->g_mop_prf >> NG_RPL_GROUNDED_SHIFT;
+    dodag->prf = dio->g_mop_prf & NG_RPL_PRF_MASK;
+
+    ng_rpl_parent_t *parent = NULL;
+
+    if (!ng_rpl_parent_add_by_addr(dodag, src, &parent) && (parent == NULL)) {
+        DEBUG("RPL: Could not allocate new parent.\n");
+        if (dodag->parents == NULL) {
+            ng_rpl_dodag_remove(dodag);
+        }
+        return;
+    }
+    else if (parent) {
+        trickle_increment_counter(&dodag->trickle);
+    }
+
+    parent->rank = byteorder_ntohs(dio->rank);
+
+    ng_rpl_parent_update(dodag, parent);
+
+    if (parent->state != 0) {
+        if (dodag->parents && (parent == dodag->parents) && (parent->dtsn != dio->dtsn)) {
+            ng_rpl_delay_dao(dodag);
+        }
+        parent->dtsn = dio->dtsn;
+
+        len -= (sizeof(ng_rpl_dio_t) + sizeof(ng_icmpv6_hdr_t));
+        _parse_options(dodag, (ng_rpl_opt_t *)(dio + 1), len, NULL);
+    }
+
+    return;
+}
+
+void _dao_fill_target(ng_rpl_opt_target_t *target, ng_ipv6_addr_t *addr)
+{
+    target->type = NG_RPL_OPT_TARGET;
+    target->length = sizeof(target->flags) + sizeof(target->prefix_length) + sizeof(target->target);
+    target->flags = 0;
+    target->prefix_length = 128;
+    target->target = *addr;
+    return;
```

Comment:
Why not? (: It indicates visually that you reached the end of the function..
I have no strong opinion on this though, I can remove it if desired